src/CoreWCF.WebHttp/src/CoreWCF/Channels/WebMessageEncodingBindingElement.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Text;
+using System.Xml;
+
+namespace CoreWCF.Channels
+{
+    public sealed class WebMessageEncodingBindingElement : MessageEncodingBindingElement//, IWsdlExportExtension
+    {
+        private WebContentTypeMapper _contentTypeMapper;
+
+        private int _maxReadPoolSize;
+        private int _maxWritePoolSize;
+        private XmlDictionaryReaderQuotas _readerQuotas;
+        private Encoding _writeEncoding;
+
+        public WebMessageEncodingBindingElement() : this(TextEncoderDefaults.Encoding)
+        {
+        }
+
+        public WebMessageEncodingBindingElement(Encoding writeEncoding)
+        {
+            if (writeEncoding == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(writeEncoding));
+            }
+
+            TextEncoderDefaults.ValidateEncoding(writeEncoding);
+            _maxReadPoolSize = EncoderDefaults.MaxReadPoolSize;
+            _maxWritePoolSize = EncoderDefaults.MaxWritePoolSize;
+            _readerQuotas = new XmlDictionaryReaderQuotas();
+            EncoderDefaults.ReaderQuotas.CopyTo(_readerQuotas);
+            _writeEncoding = writeEncoding;
+        }
+
+        WebMessageEncodingBindingElement(WebMessageEncodingBindingElement elementToBeCloned)
+            : base(elementToBeCloned)
```

Comment:
Accessibility modifier

Reply:
This is addressed in dffb7cb6fa9e55743dabb751b7ce83811f393522.